apis/essentialcontacts/v1alpha1/contact_identity.go

```diff
@@ -100,13 +100,17 @@ func NewContactIdentity(ctx context.Context, reader client.Reader, obj *Essentia
 	}
 
 	// Get desired ID
+	// This resource has a server-generated ID. This means user should not know
+	// the ID before the resource is created, and 'metadata.name' won't be used
+	// as the default resource ID. So empty value for 'spec.resourceID' should
+	// also be valid:
+	// 1. When 'spec.resourceID' is not set or set to an empty value, the
+	//    intention is to create the resource.
+	// 2. When 'spec.resourceID' is set, the intention is to acquire an existing
+	//    resource.
+	//    2.1. When 'spec.resourceID' is set but the corresponding GCP resource
+	//         is not found, then it is a real error.
```

Comment:
Personally I'd prefer not having this comment because it misses a case that if the GCP service supports both service-generated-id or user specified id.